proto/regen/ecocredit/v1/state.proto

```diff
@@ -12,40 +12,44 @@ option go_package = "github.com/regen-network/regen-ledger/x/ecocredit/core";
 message CreditType {
   option (cosmos.orm.v1alpha1.table) = {
     id : 1,
-    primary_key : {fields : "abbreviation"}
+    primary_key : {fields : "row_id", auto_increment : true}
+    index : {fields : "abbrev", unique : true}
     index : {id : 1, fields : "name", unique : true}
   };
 
-  // abbreviation is a 1-3 character uppercase abbreviation of the CreditType
-  // name, used in batch denominations within the CreditType. It must be unique.
-  string abbreviation = 1;
+  // row_id is the numeric table row identifier of the credit type.
+  uint64 row_id = 1;
 
-  // the type of credit (e.g. carbon, biodiversity, etc)
-  string name = 2;
+  // abbrev is a 1-3 character uppercase abbreviation of the CreditType name,
+  // used in batch denominations within the CreditType. It must be unique.
+  string abbrev = 2;
 
-  // the measurement unit (e.g. kg, ton, etc)
-  string unit = 3;
+  // name is the name of the credit type (e.g. carbon, biodiversity, etc).
+  string name = 3;
 
-  // the decimal precision
-  uint32 precision = 4;
+  // unit is the measurement unit of the credit type (e.g. kg, ton, etc).
+  string unit = 4;
+
+  // precision is the decimal precision of the credit type.
+  uint32 precision = 5;
 }
 
 // ClassInfo represents the high-level on-chain information for a credit class.
 message ClassInfo {
   option (cosmos.orm.v1alpha1.table) = {
     id : 2,
-    primary_key : {fields : "id", auto_increment : true}
+    primary_key : {fields : "row_id", auto_increment : true}
     index : {id : 1 fields : "name", unique : true}
     index : {id : 2 fields : "admin"}
     index : {id : 3 fields : "credit_type"}
   };
 
-  // id is the unique ID of credit class.
-  uint64 id = 1;
+  // row_id is the numeric table row identifier of the credit class.
+  uint64 row_id = 1;
```

Comment:
How about something more cryptic like dbid, rowid, or dbrow? To make it less human readable

Reply:
I think `rowid` makes the most sense out of those options. The type would be `Rowid` then rather than `RowId` though.

Reply:
How about db_key?

Reply:
When I see `db_id` or `db_key` I think the id for the database or the key for the database, not an item in the database. I still think `row_id` makes the most sense to me but I don't have a strong preference.

Reply:
I'm just trying to find a way we can avoid `id` which is used in other places like `projet_id`

Reply:
That makes sense. We can think on it some more. I do like the idea of `key` but `row_key` seems a bit odd.

Reply:
What about `pk` for primary key?

Reply:
That would probably get confusing though with `BatchBalance` that used the `pk` of the credit batch but the primary key for the `BatchBalance` table is actually `address,row_id`.

Reply:
why you want to make it cryptic? If it's an identifier then I think `id` is the right name. Since here we have other `id` field, maybe we can call this `seq` or `key`?

`pk` might be confusing.

Reply:
I think cryptic because the user should never be exposed to this. This field is only meant to be used internally.

I think we should be consistent with whatever we choose, i.e. not use `id` for some tables and `key` or something else for others. I would prefer `key` over `seq` because we use `next_sequence` in `ClassSequence`, `ProjectSequence` and `BatchSequence` and each of those would also need a numeric identifier.

Reply:
This field is only meant to be used internally. I think cryptic because the user should never be exposed to this identifier so we would make that more explicit by making it cryptic.

I think we should be consistent with whatever we choose, i.e. not use `id` when we can and `key` or something else when we can't. I would prefer `key` over `seq` because we use `next_sequence` in `ClassSequence`, `ProjectSequence` and `BatchSequence` and each of those would also need a numeric identifier.

Reply:
I've updated to `row_key` for now and will start implementing but happy to change if there is another preference.